### ISSUE TITLE
Leftovers from signal rework

### ISSUE THREAD

**Issue by boryspoplawski** [oscarlab/graphene#2140]
## Description of the problem
To finish signal rework (#2090) following TOODs/issues need to be fixed:
- initial, Pal allocated stack can be reused as LibOS syscall emulation stack,
- revisit locking around each thread->field - since there is no nested syscall calling, some might be accessed lockless,
- support SA_AUTODISARM,
- revisit PAL-Linux-SGX signal handling,
- remove `LEAVE_PAL_CALL` and friends,
- remove DkRaiseFailure (but this needs a PAL API rework - to return errors directly).

**Comment by dimakuv** [comment]
> remove DkRaiseFailure (but this needs a PAL API rework - to return errors directly).

Do you mean similar to syscalls in UNIX? Return value of `long`, negative means error code?

**Comment by boryspoplawski** [comment]
Yes, but I would like to also **not** return success values in there - I would prefer to have out-arguments to return them. Anyways, that's not something I'll just submit PR on, needs some discussion (probably on some meeting in the near future).

**Comment by donporter** [comment]
I am fully in favor of out argument.
